src/teleop/gui/src/components/GenerateReport.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+  <div>
+    <button @click="download_csv_file(spectral_data)">Generate Report</button>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    spectral_data: {
+      type: Array,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      csvFileData: [],
+    };
+  },
+  methods: {
+    download_csv_file: function (spectral_data) {
+      const time = new Date(Date.now());
+      const timeString =
+        time.toTimeString().substring(0, 17) + " " + time.toDateString();
+      this.csvFileData = [
```

Comment:
I think this could be cleaned up a bit with a map or forEach function. Or we could maybe look into packages for managing csv data.~~[This package](https://csv.js.org/) looks promising, but it is probably a later project than for this PR.~~ Sike this is a Node package never mind.